vortex-array2/src/lib.rs

```diff
@@ -0,0 +1,93 @@
+#![allow(dead_code)]
+
+pub mod compute;
+mod context;
+mod data;
+pub mod encoding;
+mod implementation;
+mod metadata;
+mod primitive;
+mod ree;
+mod validity;
+mod view;
+
+use std::fmt::Debug;
+
+pub use context::*;
+pub use data::*;
+pub use implementation::*;
+pub use metadata::*;
+pub use validity::*;
+pub use view::*;
+
+use crate::compute::ArrayCompute;
+
+#[derive(Debug, Clone)]
+pub enum Array<'v> {
+    Data(ArrayData),
+    DataRef(&'v ArrayData),
+    View(ArrayView<'v>),
+}
+
+pub trait ToArray {
+    fn to_array(&self) -> Array;
```

Comment:
when is this not a clone? Should we make this `fn to_array(self) -> Array` if you always have to clone it in the impl?